src/AlgebraicGeometry/RationalPoint/AffineRationalPoint.jl

```diff
@@ -0,0 +1,171 @@
+@doc raw"""
+    AffineRationalPoint{CoeffType<:RingElem, ParentType<:AbsSpec}
+
+A $k$-rational point ``P`` of an affine scheme ``X``.
+
+We refer to ``X`` as the parent of ``P``.
+
+# Examples
+```jldoctest
+julia> A2 = affine_space(GF(2), [:x, :y]);
+Rational point
+  of Affine 2-space over GF(2) with coordinates [x, y]
+  with coordinates (1, 0)
+
+julia> A2([1, 0])
+
+julia> (x, y) = coordinates(A2);
+
+julia> X = algebraic_set(x*y);
+
+julia> X([1,0])
+Rational point
+  of V(x*y)
+  with coordinates (1, 0)
+
+```
+"""
+struct AffineRationalPoint{S<:RingElem, T<:AbsSpec} <: AbsAffineRationalPoint{S, T}
+  # not mutable since this should be efficient
+  coordinates::Vector{S}
+  parent::T
+  function AffineRationalPoint(parent::T, coordinates::Vector{S}; check::Bool=true) where {S <: RingElem, T<:AbsSpec}
+    r = new{S, T}(coordinates, parent)
+    @check begin
+      n = dim(ambient_space(parent))
+      k = base_ring(parent)
+      n == length(coordinates) || error("$(coordinates) must have length $(n)")
```

Comment:
Is dim(ambient_space(parent)) safe in the localized setting or should we explicitly exclude (certain types of) localization? 
( The example which came to my mind is: localization of k[x,y,z] at the complement of the prime ideal generated by [x,y] )

Reply:
`ambient_space` should work always. In the example you mentioned I double checked.

Reply:
I tested several corner cases and I am content, too.